content-api/content-actors/src/main/scala/org/sunbird/content/actors/EventActor.scala

```diff
@@ -79,4 +84,51 @@ class EventActor @Inject()(implicit oec: OntologyEngineContext, ss: StorageServi
     node
   }
 
+  def joinEvent(request: Request): Future[Response] = {
+    val responseSchemaName: String = request.getContext.getOrDefault(ContentConstants.RESPONSE_SCHEMA_NAME, "").asInstanceOf[String]
+    val fields: util.List[String] = JavaConverters.seqAsJavaListConverter(request.get("fields").asInstanceOf[String].split(",").filter(field => StringUtils.isNotBlank(field) && !StringUtils.equalsIgnoreCase(field, "null"))).asJava
+    request.getRequest.put("fields", fields)
+    DataNode.read(request).map(node => {
+      val metadata: java.util.Map[String, AnyRef] = NodeUtil.serialize(node, fields, node.getObjectType.toLowerCase.replace("image", ""), request.getContext.get("version").asInstanceOf[String])
+      metadata.put("identifier", node.getIdentifier.replace(".img", ""))
+      val response: Response = ResponseHandler.OK
+
+      request.getRequest.remove("mode")
+      request.getRequest.remove("fields")
+      request.getRequest.putAll(metadata)
+      request.getRequest.remove("status")
+      request.setOperation("updateContent")
+      request.getContext.put("identifier", request.getRequest.get("identifier"))
+      val onlineProvider = request.getRequest.getOrDefault("onlineProvider", "BigBlueButton").asInstanceOf[String]
+      request.getRequest.put("onlineProvider", onlineProvider)
+
+      val meetingRequest = new Meeting()
+      meetingRequest.setMeetingID(request.getRequest.getOrDefault("identifier", java.util.UUID.randomUUID.toString).asInstanceOf[String])
+      meetingRequest.setName(request.getRequest.getOrDefault("name", "Test").asInstanceOf[String])
+      val providerApiObject = if (StringUtils.equalsIgnoreCase(onlineProvider, "BigBlueButton")) {
+        new BbbApi()
+      } else {
+        // TODO : Set response of Meeting URL for other onlineProviders
+        new BbbApi()
+      }
+
+      if (providerApiObject.deferEventCreation()) { // Creating Meeting if deferred Event creation, and updating Event with Meeting details
+        providerApiObject.createMeeting(meetingRequest)
+        if (meetingRequest.getShouldUpdate) {
+          request.getRequest.put("onlineProviderData", meetingRequest)
+          update(request) // Here, this is update event actor is called by making request identical for Update call, but event is not getting updated
```

Comment:
The `update` method here will return a Future[Response]. This will be an asynchronous execution. We are ignoring the response of this call and returning a success response always. Please handle this using `map` on Future response.

Reply:
Okay, thank you sir, working on it